finance_dl/amazon.py

```diff
@@ -264,8 +265,22 @@ def get_source():
                 source = self.driver.page_source
                 if 'Grand Total:' in source:
                     return source
+                if 'Final Details for Order' not in source:
```

Comment:
Here the logic can be confusing.
Previously `get_source` returned the source when it identified it was on an invoice.
Now, in addition to doing that, it will return the source when it identifies that it is on an intermediate pre-invoice page that Amazon Fresh orders lead to from the Orders page.
By returning the source it allows for another retry.

Reply:
Rather than checking that "Final Details for Order" is not on the page (which could indicate various kinds of failure), is there some reliable identifier on the intermediate Amazon Fresh pages that we can use to positively identify that case?

Reply:
Indeed I found some a more Fresh-specific identifier. No idea about reliability though as it's a throwaway phrase they could easily reword/remove in the future.